hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/recon/TestRecon.java

```diff
@@ -0,0 +1,385 @@
+      + "/api/containers";
+  private String fileSizeCountURL = "http://" + reconHTTPAddress
+      + "/api/utilization";
+  private String taskStatusURL = "http://" + reconHTTPAddress
+      + "/api/task/status";
+
+  @BeforeClass
+  public static void init() throws Exception {
+    dir = GenericTestUtils.getRandomizedTestDir();
+    conf = new OzoneConfiguration();
+    conf.set(HddsConfigKeys.OZONE_METADATA_DIRS, dir.toString());
+    ugi = UserGroupInformation.getCurrentUser();
+
+
+    int socketTimeout = (int) conf.getTimeDuration(
+        RECON_OM_SOCKET_TIMEOUT, RECON_OM_SOCKET_TIMEOUT_DEFAULT,
+        TimeUnit.MILLISECONDS);
+    int connectionTimeout = (int) conf.getTimeDuration(
+        RECON_OM_CONNECTION_TIMEOUT,
+        RECON_OM_CONNECTION_TIMEOUT_DEFAULT, TimeUnit.MILLISECONDS);
+    int connectionRequestTimeout = (int)conf.getTimeDuration(
+        RECON_OM_CONNECTION_REQUEST_TIMEOUT,
+        RECON_OM_CONNECTION_REQUEST_TIMEOUT_DEFAULT, TimeUnit.MILLISECONDS);
+    RequestConfig config = RequestConfig.custom()
+        .setConnectTimeout(socketTimeout)
+        .setConnectionRequestTimeout(connectionTimeout)
+        .setSocketTimeout(connectionRequestTimeout).build();
+
+    cluster =  MiniOzoneCluster.newBuilder(conf).build();
+    cluster.waitForClusterToBeReady();
+    metadataManager = cluster.getOzoneManager().getMetadataManager();
+
+    cluster.getStorageContainerManager().exitSafeMode();
+    reconUtils = cluster.getReconServer().getInjector().getInstance(
+        ReconUtils.class);
+
+    // initialize HTTPClient
+    httpClient = HttpClientBuilder
+        .create()
+        .setDefaultRequestConfig(config)
+        .build();
+
+    pauseInterval = conf.getTimeDuration(
+        conf.get(RECON_OM_SNAPSHOT_TASK_INTERVAL),
+        conf.get(RECON_OM_SNAPSHOT_TASK_INTERVAL_DEFAULT),
+        TimeUnit.MILLISECONDS);
+  }
+
+  @AfterClass
+  public static void shutdown() {
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+  }
+
+  /**
+   * Returns a {@link CloseableHttpClient} configured by given configuration.
+   * If conf is null, returns a default instance.
+   *
+   * @param url        URL
+   * @return a JSON String Response.
+   */
+  private String makeHttpCall(String url)
+      throws IOException {
+    HttpGet httpGet = new HttpGet(url);
+    HttpResponse response = httpClient.execute(httpGet);
+    int errorCode = response.getStatusLine().getStatusCode();
+    HttpEntity entity = response.getEntity();
+
+    if ((errorCode == HTTP_OK) || (errorCode == HTTP_CREATED)) {
+      return EntityUtils.toString(entity);
+    }
+
+    if (entity != null) {
+      throw new IOException("Unexpected exception when trying to reach Ozone " +
+          "Manager, " + EntityUtils.toString(entity));
+    } else {
+      throw new IOException("Unexpected null in http payload," +
+          " while processing request");
+    }
+  }
+
+  @Test
+  public void testReconServer() throws Exception {
```

Comment:
For testing Recon restart, I would have to initialize keys and check for reprocess which will be done as part of testing reprocess and delta updates for Recon, so to avoid re-testing the same, I opted for consolidating the whole scenario together.

Reply:
Ok, we can keep it as 1 test then. But, I have added a suggestion in my review that will decrease the number of lines in this test.